workers/deployment_manager/DeploymentManager/Worker/SpatialOSReceiveHandler.cs

```diff
@@ -0,0 +1,73 @@
+using Improbable.Worker;
+using Deployment = Improbable.SpatialOS.Deployment.V1Alpha1.Deployment;
+
+namespace DeploymentManager
+{
+    public class SpatialOSReceiveHandler
+    {
+        private readonly Dispatcher dispatcher = new Dispatcher();
+        private readonly Connection connection;
+        private readonly string deploymentId;
+        private readonly string projectName;
+
+        public SpatialOSReceiveHandler(Connection connection, string deploymentId, string projectName)
+        {
+            this.connection = connection;
+            this.deploymentId = deploymentId;
+            this.projectName = projectName;
+            SetupHandlers();
+        }
+
+        public void ProcessOps()
+        {
+            using (var opList = connection.GetOpList(0))
+            {
+                dispatcher.Process(opList);
+            }
+        }
+
+        private void SetupHandlers()
+        {
+            dispatcher.OnLogMessage(OnLogMessage);
+            dispatcher.OnAddComponent<Improbable.Session.Session>(OnComponentAdded);
+            dispatcher.OnComponentUpdate<Improbable.Session.Session>(OnComponentUpdate);
+        }
+
+        private void OnLogMessage(LogMessageOp message)
+        {
+            Log.Print(message.Level, message.Message, connection);
+        }
+
+
+        private void OnComponentAdded(AddComponentOp<Improbable.Session.Session> componentAddOp)
+        {
+            Log.Print(LogLevel.Info, "receiving component update", connection);
```

Comment:
The log message is wrong, but you probably want to remove this and all the other debug prints, anyway.